src/scala/ly/stealth/mesos/kafka/Scheduler.scala

```diff
@@ -87,8 +87,14 @@ object Scheduler extends org.apache.mesos.Scheduler {
         "host.name" -> offer.getHostname
       )
 
-      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0)
-        defaults += ("listeners" -> s"PLAINTEXT://:${reservation.port}")
+      if (HttpServer.kafkaVersion.compareTo(new Version("0.9")) >= 0) {
+        if (broker.bindAddress != null) {
+          defaults += ("listeners" -> s"PLAINTEXT://${offer.getHostname}:${reservation.port}")
```

Comment:
The stack trace you posted is the existing call to `result.put("host.name", bindAddress.resolve())`.  My point was you need to add the code that sets the listener address in the same place as that existing call so it runs on the executor and not the scheduler.